vms/job/tests/test_searchJob.py

```diff
@@ -0,0 +1,457 @@
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.common.by import By
+
+# Django
+from django.contrib.staticfiles.testing import LiveServerTestCase
+
+# local Django
+from pom.pages.authenticationPage import AuthenticationPage
+from pom.pages.jobSearchPage import JobSearchPage
+from shift.utils import create_admin, create_event_with_details, register_event_utility, create_job_with_details
+
+class SearchJob(LiveServerTestCase):
+    """
+    SearchJob class contains tests to check '/job/list/' view.
+    Choices of parameters contains
+    - Name
+    - Start Date
+    - End Date
+    - City
+    - State
+    - Country
+    - Event
+    Class contains 7 tests to check each parameter separately and also to check
+    if a combination of parameters entered, then intersection of all results is
+    obtained.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Method to initiate class level objects.
+        This method initiates Firefox WebDriver, WebDriverWait and
+        the corresponding POM objects for this Test Class
+        """
+        cls.driver = webdriver.Firefox()
+        cls.driver.implicitly_wait(5)
+        cls.driver.maximize_window()
+        cls.search_page = JobSearchPage(cls.driver)
+        cls.authentication_page = AuthenticationPage(cls.driver)
+        cls.wait = WebDriverWait(cls.driver, 10)
+        super(SearchJob, cls).setUpClass()
+
+    def setUp(self):
+        """
+        Method consists of statements to be executed before
+        start of each test
+        """
+        create_admin()
+        self.login_admin()
+        self.wait_for_home_page()
+
+    def tearDown(self):
+        """
+        Method contains statements to be executed at the end of
+        each test.
+        """
+        self.authentication_page.logout()
+
+    @classmethod
+    def tearDownClass(cls):
+        """
+        Class method to quit the Firefox WebDriver session after
+        execution of all tests in class.
+        """
+        cls.driver.quit()
+        super(SearchJob, cls).tearDownClass()
+
+    def login_admin(self):
+        """
+        Utility function to login an admin user to perform all tests.
+        """
+        self.authentication_page.server_url = self.live_server_url
+        self.authentication_page.login({
+            'username': 'admin',
+            'password': 'admin'
+        })
+
+    def wait_for_home_page(self):
+        self.wait.until(
+            EC.presence_of_element_located(
+                (By.XPATH, "//h1[contains(text(), 'Volunteer Management System')]")
+            )
+        )
+
+    def test_job_name_field(self):
+        """
+        Test search results for job name
+        """
+        search_page = self.search_page
+        search_page.live_server_url = self.live_server_url
+
+        event_obj = register_event_utility()
+
+        credentials_1 = ['job-name', '2050-06-10', '2050-06-11', 'job-description', event_obj]
```

Comment:
This is creating redundancy, since all the tests are using this we should create a static function to do this job. You can see other tests have this `create_valid_event()` `create_valid_job()` etc.

Reply:
I agree with @Monal5031 on this. Won't adding these functions in `setUp()` function work?